File: intellij/src/main/kotlin/com/wquasar/codeowners/lens/core/CodeOwnerRule.kt
package com.wquasar.codeowners.lens.core

internal data class CodeOwnerRule(
    val pattern: String,
    var owners: List<String>,
    val lineNumber: Int,
) {
    companion object {
        fun fromCodeOwnerLine(lineNumber: Int, lineInfo: List<String>) =
            CodeOwnerRule(lineInfo[0], lineInfo.drop(1).sorted(), lineNumber)
    }
}

File: vscode/src/codeOwnerRule.ts
export interface CodeOwnerRule {
    pattern: string;
    owners: string[];
    lineNumber: number;
}

export function parseCodeOwnerLine(lineNumber: number, lineInfo: string[]): CodeOwnerRule | null {
    if (lineInfo.length < 2) {
        return null;
    }
    return {
        pattern: lineInfo[0],
        owners: lineInfo.slice(1).sort(),
        lineNumber
    };
}
